lib/cea/cea608_data_channel.js

```diff
@@ -0,0 +1,720 @@
+
+    // Execute the PAC.
+    const buf = this.curbuf;
+    if (this.type_ === shaka.cea.Cea608DataChannel.CaptionType.TEXT) {
+      row = buf.getRow(); // Don't change row if in text mode.
+    } else if (this.type_ === shaka.cea.Cea608DataChannel.CaptionType.ROLLUP) {
+      if (row !== buf.getRow()) { // Move entire scroll window to a new base.
+        const oldtoprow = 1 + buf.getRow() - buf.getScrollSize();
+        const newtoprow = 1 + row - buf.getScrollSize();
+
+        // Shift up the scroll window.
+        buf.moveRows(newtoprow, oldtoprow, buf.getScrollSize());
+
+        // Clear everything outside of the new scroll window.
+        buf.resetRows(0, newtoprow - 1);
+        buf.resetRows(row + 1,
+            shaka.cea.Cea608Memory.CC_ROWS - row);
+      }
+    }
+    buf.setRow(row);
+
+    this.curbuf.setUnderline(underline);
+    this.curbuf.setItalics(italics);
+    this.curbuf.setTextColor(textColor);
+
+    // Clear the background color, since new row (PAC) should reset ALL styles.
+    this.curbuf.setBackgroundColor(shaka.cea.Cea608Memory.DEFAULT_BG_COLOR);
+  }
+
+  /**
+   * Mid-Row control code handler.
+   * @param {!number} b2 Byte #2.
+   * @private
+   */
+  controlMidrow_(b2) {
+    // Clear all pre-existing midrow style attributes.
+    this.curbuf.setUnderline(false);
+    this.curbuf.setItalics(false);
+    this.curbuf.setTextColor(shaka.cea.Cea608Memory.DEFAULT_TXT_COLOR);
+
+    // Mid-row attrs use a space.
+    this.curbuf.addChar(
+        shaka.cea.Cea608Memory.CharSet.BASIC_NORTH_AMERICAN, ' '.charCodeAt(0));
+
+    let textColor = shaka.cea.Cea608Memory.DEFAULT_TXT_COLOR;
+    let italics = false;
+
+    // Midrow codes set underline on last (LSB) bit.
+    const underline = (b2 & 0x01) === 0x01;
+
+    // b2 has the form |P|0|1|0|STYLE|U|
+    textColor = shaka.cea.Cea608DataChannel.TEXT_COLORS[(b2 & 0xe) >> 1];
+    if (textColor === 'white_italics') {
+      textColor = 'white';
+      italics = true;
+    }
+
+    this.curbuf.setUnderline(underline);
+    this.curbuf.setItalics(italics);
+    this.curbuf.setTextColor(textColor);
+  }
+
+  /**
+   * Background attribute control code handler.
+   * @param {!number} b1 Byte #1
+   * @param {!number} b2 Byte #2.
+   * @private
+   */
+  controlBackgroundAttribute_(b1, b2) {
+    let backgroundColor = shaka.cea.Cea608Memory.DEFAULT_BG_COLOR;
+    if ((b1 & 0x07) === 0x0) {
+      // If background provided, last 3 bits of b1 are |0|0|0|. Color is in b2.
+      backgroundColor = shaka.cea.Cea608DataChannel.BG_COLORS[(b2 & 0xe) >> 1];
+    }
+    this.curbuf.setBackgroundColor(backgroundColor);
+  }
+
+  /**
+   * The Cea608DataChannel control methods implement all CC control operations.
+   * @param {!shaka.cea.Cea608DataChannel.CcData} ccPacket
+   * @private
+   */
+  controlMiscellaneous_(ccPacket) {
+    const b2 = ccPacket.cc2;
+    const pts = ccPacket.pts;
+    switch (b2) {
+      case shaka.cea.Cea608DataChannel.MiscCmd_.RCL:
```

Comment:
You can alias MiscCmd at the top of the function:

```js
const MiscCmd = shaka.cea.Cea608DataChannel.MiscCmd_;
```

Then shorten these case statements.